### PR TITLE
Allow chaining after setRenderer

### PR DESCRIPTION
**Pull request by JSteunou**
Because one day I get bitten by `export default Mn.View.extend(...).setRenderer(...)`

### CODE REVIEW

**Comment by paulfalgout** [comment]
bleh chaining..  I guess it is a convention we should follow.. however in that case `setDomApi` should do the same.  Also.. as silly as it is, we should test this

**Comment by JSteunou** [comment]
So maybe I should include `setDomApi` chaining in this PR along with testing ?

**Comment by rafde** [comment]
@paulfalgout @JSteunou this conflicts with https://github.com/marionettejs/backbone.marionette/issues/3242

**Comment by JSteunou** [comment]
arrgh :(

**Comment by paulfalgout** [comment]
I don't see how it conflicts?  The API for setting the function isn't the same as being a function meant to be overridden.  I don't think anyone should overwrite `setRenderer` or `setDomApi`

**Comment by rafde** [comment]
What is meant to be overridden it vague. The time I know something isn't meant to be overridden is when property/function name has an `_` in front of it. All others are safe to override.
Clarification for https://github.com/marionettejs/backbone.marionette/issues/3242 is probably needed, then.

**Comment by paulfalgout** [comment]
I think what is meant to be overridden is documented.  There will never be a case of a `_`'d function that returns `this`.  Only public API returns `this`, and those public API that do return `this` should not be overridden.  This is because those function call internally private API and could not be reproduced otherwise. Otherwise the original issue should have said "never return `this` because all public methods are overriddable" but I do not think that is the case at all.  
The `_` only distinguishes between methods you should call directly or override without fear of breaking for minor versions, and methods that you should _not_ call _or_ override because their interface and even existence is not guaranteed within a major version.

**Comment by blikblum** [comment]
setRenderer and setDomApi are class methods unlike thebl cited in other issue

**Comment by scott-w** [comment]
I agree with @paulfalgout that we should add a test for the `return this` and formalise the definition.

I assume this goes in a 3.5 release?

**Comment by paulfalgout** [comment]
Anyone up for saying "`setRenderer` and `setDomApi` were not returning the prototype as expected" and add this as a bugfix for `3.4.2`?

File: test/unit/config/dom.js [hunk 1]
```diff
@@ -22,6 +23,13 @@ chai.use(function(_chai, utils) {
 
 
 describe('DomApi', function() {
+  describe('#setDomApi', function() {
+    it('should return the current class', function() {
+      const MyObject = Backbone.Marionette.Object.extend({}, {setDomApi: setDomApi});
```
Comment (paulfalgout) [inline test/unit/config/dom.js]: it'd be nice to do unit tests without having to do a build in the future.. so it'd be nice not to use the global `Backbone` or `Marionette`.  In this case though I think you could just 
```js
const MyObject = function() { };
MyObject.setDomApi = setDomApi;
```

File: test/unit/config/dom.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import $ from 'jquery';
 import _ from 'underscore';
 import DomApi from '../../../src/config/dom';
+import { setDomApi } from '../../../src/config/dom';
```
Comment (paulfalgout) [inline test/unit/config/dom.js]: `import DomApi, { setDomApi } from '../../../src/config/dom';`

**Comment by JSteunou** [comment]
Good call @paulfalgout, I pushed the changes

**Pull request merged**